subprojects/core-api/src/main/java/org/gradle/process/ExecException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process;
+
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * An exception thrown when an error occurs while executing a process.
+ *
+ * @since 9.0
+ */
+@NullMarked
+@SuppressWarnings("deprecation")
+public class ExecException extends org.gradle.process.internal.ExecException {
```

Comment:
❓ How about we call this `ExecutionException`, or even `ProcessExecutionException` to distinguish it from other executions?

The term "exec" is used internally to refer to our `Exec` infrastructure, but this exception belongs to the somewhat larger scope of "process management". I think it's better to make the name more general.

As a slight added bonus, it would also make the weird FQCN references unnecessary in code that lives in the internal package.